parser/openfoodfacts_taxonomy_parser/parser.py

```diff
@@ -175,22 +176,37 @@ def header_harvest(self, filename):
 
     def entry_end(self, line, data):
         """Return True if the block ended"""
-        if "stopwords" in line or "synonyms" in line or not line:
-            # can be the end of an block or just 2 line separators,
+        # stopwords and synonyms are one-liner, entries are separated by a blank line
+        if line.startswith("stopwords") or line.startswith("synonyms") or not line:
+            # can be the end of an block or just additional line separator,
             # file_iter() always end with ''
             if data["id"]:  # to be sure that it's an end
                 return True
         return False
 
     def remove_separating_line(self, data):
+        """
+        To remove the one separating line that is always there,
+        between synonyms part and stopwords part and before each entry
+        """
+        # first, check if there is at least one preceding line
         if data["preceding_lines"]:
-            if "synonyms" in data["id"]:
+            if data["id"].startswith("synonyms"):
+                # it's a synonyms block,
+                # if the previous block is a stopwords block,
+                # there is at least one separating line
                 if "stopwords" in self.is_before:
                     data["preceding_lines"].pop(0)
-            elif "stopwords" in data["id"]:
+
+            elif data["id"].startswith("stopwords"):
+                # it's a stopwords block,
+                # if the previous block is a synonyms block,
+                # there is at least one separating line
                 if "synonyms" in self.is_before:
                     data["preceding_lines"].pop(0)
+
             else:
+                # it's an entry block, there is always a separating line
                 data["preceding_lines"].pop(0)
```

Comment:
I think your assumption here are too much based upon what you have seen so far, without any guarantee. There is no guarantee that we have new lines if we change type and so on. Remember that taxonomies are formatted by humans.
You need at list to check that `data["preceding_lines"][0]` is empty.

So I really think my proposal is the good one.
see: https://github.com/openfoodfacts/taxonomy-editor/pull/18#discussion_r932470399
If in rare cases it adds or remove a blank line, this is ok, I mean we do not lose any important information.